rules/library.bzl

```diff
@@ -940,8 +941,8 @@ def apple_library(
                 "@build_bazel_rules_ios//:swift_disable_import_underlying_module": [],
                 "//conditions:default": ["-import-underlying-module"] if not feature_names.swift_disable_import_underlying_module in features else [],
             })
+            swiftc_inputs += [module_map]
```

Comment:
Does this already get added for the extended modulemap below? Or is that overridden 👀

Reply:
This `module_map` is passed to framework_vfs_overlay_name_swift, while the one in extended modulemap below is different and passed to framework_vfs_overlay_name and objc_libname. I think it's unnecessary to add the extended modulemap to swiftc_inputs.

(BTW, the code is really hard to reason. We need to clean it up)

Reply:
@gyfelton Could you test deleting line 965 to see if it still works as expected?

Reply:
Yeah agreed on it being hard to reason about. I'm happy to merge whatever has the least amount of changes, thinking about it more im not sure extended module map should be an input so let's try deleting it. 

It might be best to create a sandbox test ci job and merge these together

Reply:
So the extended modulemap has two major diff:
1. location: the added module map has umbrella header next to it, but the extended one is at root
<img width="562" alt="Screenshot 2024-08-19 at 9 43 48 AM" src="https://github.com/user-attachments/assets/9cb73ab5-0004-45c4-ac4b-e6bb11bc3a82">
2. content:
extended one has the following:
```
framework module SwiftLibrary {
    umbrella header "SwiftLibrary-umbrella.h"

    export *
    module * { export * }
}

module SwiftLibrary.Swift {
    header "SwiftLibrary-Swift.h"
    requires objc
}
```
The SwiftLibrary.Swift module is the net addition to the original module map

deleting the adding of extended module map works for my case but i am not sure about the implication of removing it.
I will add a TODO here for now and if we all agree on the future plan, i can come back to this esp. during the time when i need to have rules_ios to fully pass CI in sandbox.  WDYT?

Reply:
strike that, that is needed or we got  `-swift.h` file not found error in our repo's compilation

Reply:
> that is needed or we got -swift.h file not found error in our repo's compilation

Yes, we still need the extended modulemap for framework_vfs_overlay_name and objc_libname, but I don't think we need to add it to swiftc_inputs on line 967 because the swift compilation shouldn't need the umbrella header or -Swift.h.